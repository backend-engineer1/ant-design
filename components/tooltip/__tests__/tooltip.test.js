import React from 'react';
import { mount } from 'enzyme';
import Tooltip from '..';

describe('Tooltip', () => {
  it('check `onVisibleChange` arguments', () => {
    const onVisibleChange = jest.fn();

    const wrapper = mount(
      <Tooltip
        title=""
        mouseEnterDelay={0}
        mouseLeaveDelay={0}
        onVisibleChange={onVisibleChange}
      >
        <div>Hello world!</div>
      </Tooltip>
    );

    // `title` is empty.
    const div = wrapper.find('div').at(0);
    div.simulate('mouseenter');
    expect(onVisibleChange).not.toHaveBeenCalled();
    expect(wrapper.ref('tooltip').prop('visible')).toBe(false);

    div.simulate('mouseleave');
    expect(onVisibleChange).not.toHaveBeenCalled();
    expect(wrapper.ref('tooltip').prop('visible')).toBe(false);

    // update `title` value.
    wrapper.setProps({ title: 'Have a nice day!' });
    wrapper.simulate('mouseenter');
    expect(onVisibleChange).toHaveBeenLastCalledWith(true);
    expect(wrapper.ref('tooltip').prop('visible')).toBe(true);

    wrapper.simulate('mouseleave');
    expect(onVisibleChange).toHaveBeenLastCalledWith(false);
    expect(wrapper.ref('tooltip').prop('visible')).toBe(false);

    // add `visible` props.
    wrapper.setProps({ visible: false });
    wrapper.simulate('mouseenter');
    expect(onVisibleChange).toHaveBeenLastCalledWith(true);
    const lastCount = onVisibleChange.mock.calls.length;
    expect(wrapper.ref('tooltip').prop('visible')).toBe(false);

    // always trigger onVisibleChange
    wrapper.simulate('mouseleave');
    expect(onVisibleChange.mock.calls.length).toBe(lastCount); // no change with lastCount
    expect(wrapper.ref('tooltip').prop('visible')).toBe(false);
  });
});
