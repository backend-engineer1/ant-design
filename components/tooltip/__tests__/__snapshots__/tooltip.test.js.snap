// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip should hide when mouse leave antd disabled component Button 1`] = `
<span
  style="display: inline-block; cursor: not-allowed;"
>
  <button
    class="ant-btn"
    disabled=""
    style="pointer-events: none;"
    type="button"
  />
</span>
`;

exports[`Tooltip should hide when mouse leave antd disabled component Switch 1`] = `
<span
  style="display: inline-block; cursor: not-allowed;"
>
  <button
    aria-checked="false"
    class="ant-switch ant-switch-disabled"
    disabled=""
    role="switch"
    style="pointer-events: none;"
    type="button"
  >
    <span
      class="ant-switch-inner"
    />
  </button>
</span>
`;
